client/windows/CCreatureWindow.cpp

```diff
@@ -755,17 +759,16 @@ CStackWindow::CStackWindow(const CStackInstance * stack, bool popup)
 	init();
 }
 
-CStackWindow::CStackWindow(const CStackInstance * stack, std::function<void()> dismiss, const UpgradeInfo & upgradeInfo, std::function<void(CreatureID)> callback)
+CStackWindow::CStackWindow(const CStackInstance * stack, std::function<void()> dismiss, UpgradeInfo && upgradeInfo, std::function<void(CreatureID)> callback)
 	: CWindowObject(BORDERED),
 	info(new UnitView())
 {
 	info->stackNode = stack;
 	info->creature = stack->getCreature();
 	info->creatureCount = stack->count;
 
-	info->upgradeInfo = std::make_optional(UnitView::StackUpgradeInfo());
+	info->upgradeInfo = std::make_optional(UnitView::StackUpgradeInfo(std::move(upgradeInfo)));
```

Comment:
To be honest previous approach might have been better.

Sure, your change might result in a little bit faster code, depending on compiler behavior. But this is not performance-critical code, and it makes code a little more complex for no reason (due to now required std::move). And most of our codebase does not use this approach.

Reply:
Got rid of std::move